rollup.config.mjs

```diff
@@ -0,0 +1,71 @@
+// rollup.config.mjs
+
+import commonjs from '@rollup/plugin-commonjs';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import terser from '@rollup/plugin-terser';
+import alias from '@rollup/plugin-alias';
+
+const banner = `/* Copyright 2015-${new Date().getFullYear()} PlayCanvas Ltd */\n`;
+
+const builds = [
+    {
+        name: 'playcanvas-spine.3.6',
+        lib: 'contrib/spine-ts/build/3.6/spine-core.js'
+    },
+    {
+        name: 'playcanvas-spine.3.8',
+        lib: 'contrib/spine-ts/build/3.8/spine-core.js'
+    },
+    {
+        name: 'playcanvas-spine.4.1',
+        lib: 'src/wrapper.js'
+    }
+];
+
+/**
+ * Return all the build targets.
+ *
+ * @returns {RollupOptions | RollupOptions[]} Build targets.
+ */
+export default [
+    ...builds.map(buildDefinition => buildTarget(buildDefinition))
+];
+
+/**
+ * Return a target that rollup is supposed to build.
+ *
+ * @param {{ name, lib }} buildDefinition - The build definition.
+ * @returns {RollupOptions} One rollup target.
+ */
+function buildTarget({ name, lib }) {
+
+    const outputDir = './build/';
+
+    const outputPlugins = {
+        release: [],
+        min: [terser()]
+    };
+
+    const entries = { './wrapper.js': lib };
+    const buildPlugins = [alias({ entries }), commonjs(), nodeResolve()];
+
+    return {
+        input: 'src/plugin.js',
+        context: 'this', // remove when using ESM libs
+        output: [
+            {
+                file: `${outputDir + name}.js`,
+                format: 'cjs',
```

Comment:
Shouldn't this be `'umd'`?

Reply:
It does not have value.. yet.  That's for the rewrite.  There's no module exports yet.

- cjs will just push the processed files, you get the nearly same bundled output with treeshake
- umd will put a wrapper function with no name, no exports, that is stripped away in the minified builds.  In short, it indents text and gets removed.

It has a place in the next steps:

The next steps are generating ESM bundles for 3.6 and 3.8 from the typescript source because vendor only publishes processed typescript to npm from 4.0.  At that point, we need to remove the "context" (it needs to be top level undefined), introduce babbel to support language correct es5 and es6 targets (if translation is required), and flip this module setting accordingly.

Reply:
OK, sounds good.